gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResource.java

```diff
@@ -222,29 +180,11 @@ public IngestedApisResponse getIngestedApis(
     @Produces(MediaType.APPLICATION_JSON)
     @Permissions(
         {
-            @Permission(value = RolePermission.ENVIRONMENT_INTEGRATION, acls = { RolePermissionAction.READ }),
-            @Permission(value = RolePermission.ENVIRONMENT_API, acls = { RolePermissionAction.DELETE }),
+            @Permission(value = RolePermission.INTEGRATION_DEFINITION, acls = { RolePermissionAction.DELETE }),
+            @Permission(value = RolePermission.ENVIRONMENT_INTEGRATION, acls = { RolePermissionAction.DELETE }),
         }
     )
     public DeletedIngestedApisResponse deleteIngestedApis(@PathParam("integrationId") String integrationId) {
-        var executionContext = GraviteeContext.getExecutionContext();
-        if (
-            !hasPermission(
-                executionContext,
-                RolePermission.ENVIRONMENT_INTEGRATION,
-                GraviteeContext.getCurrentEnvironment(),
-                RolePermissionAction.READ
-            ) ||
-            !hasPermission(
-                executionContext,
-                RolePermission.ENVIRONMENT_API,
-                GraviteeContext.getCurrentEnvironment(),
-                RolePermissionAction.DELETE
-            )
-        ) {
-            throw new ForbiddenAccessException();
-        }
-
```

Comment:
We need to keep and adapt this piece of code because annotation does not allow an OR check. The spec says to allow deletion when user have `INTEGRATION_DEFINITION[DELETE]` OR `ENVIRONMENT_INTEGRATION[DELETE]` permissions

Reply:
Multiple permissions implement an `OR` logic (it's in [PermissionFilter](https://github.com/gravitee-io/gravitee-api-management/blob/7cedcac9a407c58a4af3871b2f4974a65e791f95/gravitee-apim-rest-api/gravitee-apim-rest-api-rest/src/main/java/io/gravitee/rest/api/rest/filter/PermissionsFilter.java#L74-L78)). It's seems OK, no?
I will add documentation on `Permissions` annotation

Reply:
nevermind I miss-read 🤯